### PR TITLE
Proposed way to include packages like XPlot and FSCharting 

### PR DESCRIPTION
**Pull request by npmurphy**
Bug #53 seems to be caused by incompatible .Net versions.
Bug #52 is caused by hard coding a specific version of FSCharting into IfSharp. 

This pull request is a suggestion to fix both these problems by using Paket in the installation directory.
Paket can provide the correct version of the libraries for the users framework as well as the latest versions via nuget or github. 

This also acts as a kind of light-weight plugin system. To add support for a new library one adds it to the paket.dependancies file and adds an fsx script to load the dlls and add helper functions. 

I have demonstrated the idea in this PR with XPlot.Plotly. 

The user runs `#load XPlot.Plotly.fsx` in their notebook (similar to  `%matplotlib inline` in IPython) and the namespace is populated with the functions needed to use XPlot.Plotly (`InitialiseNotebook` and `Show`).

After IfSharp installation the user should run `./paket/paket.bootstrapper.exe` from their installation directory. 
Then  `paket.exe install` to download the latest packages to their installation folder. 
In future these commands could be run automatically on the first run of `ifsharp.exe` or some "magic" `%paket` commands could be added to the IfSharp Kernel. 

I have also prepared a similar solution for FSCharting, the library has been removed from the Kernel and is added back to the kernel at runtime by an fsx script.

### CODE REVIEW

**Comment by cgravill** [comment]
This approach looks like it has a lot of potential.

When I attempt to load this with Visual Studio 2015 I get get failures for duplicate items. I've taken a look at IfSharp.Kernel.fsproj and you're doing something with custom elements:

```XML
<None Include=".\helpers\XPlot.Plotly.fsx">
      <!-- Copied in a post build event -->
      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
    </None>
```
and

```XML
<Helper Include=".\helpers\XPlot.Plotly.fsx" />
```

does this work for you on other platforms? Could you perhaps just copy the whole helpers directory instead?

**Comment by npmurphy** [comment]
It worked fine for me using FAKE, but I see in MonoDevelop that the files show up twice in the Project explorer. 
I have fixed that and I hope it also works for VS.  
Also now the paket files are put in the build output directory instead of .paket.

**Comment by cgravill** [comment]
Great, when testing it on windows I got a crash against msbuild. I was able to resolve this by updating the F# Compiler Service. I sent you a PR npmurphy/IfSharp#5

When I try running `#load "XPlot.Plotly.fsx"` it's not seeing the file. For the copy
```xml
<Copy SourceFiles="@(CopyHelpers)" DestinationFolder="$(OutDir)" />
```
should that be aiming for the IfSharpConsole output directory instead?

File: src/IfSharp.Kernel/helpers/paket.dependencies [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source https://www.nuget.org/api/v2
+
+nuget FSharp.Data.TypeProviders 
+nuget XPlot.Plotly
```
Comment (dsyme) [inline src/IfSharp.Kernel/helpers/paket.dependencies]: It's almost like you could have all of FsLab (http://fslab.org) as a dependency.  Not quite sure of the ramifications of that though!

**Comment by npmurphy** [comment]
@cgravill Yes it should. I have fixed that.

**Comment by npmurphy** [comment]
https://github.com/npmurphy/IfSharp/tree/fscharting_ripout
Provisional version of decoupling FSCharting using this method is here. 
Please try it out. 
For FSCharting.GTK I am currently modifying FSCharting.Gtk and using an unstable version of Oxyplot.

**Comment by cgravill** [comment]
OK, I'll merge this then we can look at the other improvements.

File: src/IfSharp.Kernel/helpers/paket.dependencies (hunk 1 above)
Comment (cgravill) [inline src/IfSharp.Kernel/helpers/paket.dependencies]: @dsyme I'm open including more. I checked and FsLab comes down as about 100MB.

There's a bit of question of how to let the user change versions. At the moment we've been saying they need to go edit this paket.dependencies file. Perhaps if we add helper functions to let users alter the dependencies inside the notebook? Then Paket will make changes as needed.

**Pull request merged**